atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/batch/AccumulatorStrategies.java

```diff
@@ -44,6 +46,8 @@ private AccumulatorStrategies() {}
 
         @Override
         public void accumulateResult(Map<K, V> result, Map<K, V> toAdd) {
+            Preconditions.checkArgument(Collections.disjoint(result.keySet(), toAdd.keySet()),
```

Comment:
In the general case if result is actually a multimap this check would reject things that should be valid.